### PR TITLE
Add Android lifecycles test

### PR DESCRIPTION
**Pull request by blasten**
Add Android lifecycle test, and minor clean up to android_choreographer_do_frame_test.dart 
for consistency.

Tests: Activity#onStop(), Activity#onResume(), Activity#onPause() 
from Dart perspective in  debug, profile, and release modes.

Issue: https://github.com/flutter/flutter/issues/97853

### CODE REVIEW

File: dev/devicelab/lib/tasks/android_choreographer_do_frame_test.dart [hunk 1]
```diff
@@ -18,12 +18,11 @@ const List<String> kSentinelStr = <String>[
   '==== sentinel #3 ====',
 ];
 
-// Regression test for https://github.com/flutter/flutter/issues/98973
-// This test ensures that Choreographer#doFrame finishes during application startup.
-// This test fails if the application hangs during this period.
-// https://ui.perfetto.dev/#!/?s=da6628c3a92456ae8fa3f345d0186e781da77e90fc8a64d073e9fee11d1e65
+/// Tests that Choreographer#doFrame finishes during application startup.
```
Comment (blasten) [inline dev/devicelab/lib/tasks/android_choreographer_do_frame_test.dart:21]: This is all a clean up. It should probably be in a different PR :/

**Review by keyonghan — COMMENTED** [review]
.ci.yaml and TESTOWNERS LGTM.

**Review by jason-simmons — APPROVED** [review]
